src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionService.kt

```diff
@@ -80,8 +85,46 @@ class IdentificationSessionService(
         return session
     }
 
-    fun delete(identificationSession: IdentificationSession) {
-        identificationSessionRepository.delete(identificationSession)
-        log.info("Deleted identification session. useIdSessionId=${identificationSession.useIdSessionId}")
+    @Transactional
```

Comment:
I was wondering the same for the PANSTART draft. Maybe now is the better point to ask:
- Why do we need to set transactional here?
- Why do we set it for the whole function and not only the part that actually does sth with the database?
   -  Is that because we need to keep getEidInformationa and deleteSession together transactional?

Reply:
I read again [this documentation](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#transactions) and would therefore propose to set the transaction method next to the `@Modifying` query on the repository method. The transaction is needed here since we are using a custom `@Query` and  Spring does not know ootb if it's a `readOnly` method anymore.

Reply:
Thanks. That makes sense to me 👍